48 order page

no4any opened the pull request.

no4any committed:
#48 integrated cart management

no4any committed:
#48 removed unsed use

no4any committed:
#48 fixed linting

no4any committed:
Merge branch 'main' into 48-order-page

takh86 reviewed (approved):
Es sieht gut aus

untbu commented:
This doesn't close #48. The issue is about an overview page for a completed order, while this adds some functionality to the cart.

untbu commented on packages/web/src/app/cart/cart.component.tsx:
```diff
@@ -1,18 +1,21 @@
-    const [items, setItems] = useState<ShoppingCartItem[]>(products);
+    const [items, setItems] = useState<ShoppingCartItem[]>([]);
 
     const handleQuantityChange = (id: number, delta: number) => {
```
This function updates the displayed items instead of updating the cart. After reloading, the changes are gone.

untbu commented on packages/web/src/app/cart/cart.component.tsx:
```diff
@@ -37,7 +47,9 @@ const ShoppingCart: React.FC<ShoppingCartProps> = ({ products = [] }) => {
         <main className="mx-auto max-w-2xl px-4 pt-16 pb-24 sm:px-6 lg:max-w-7xl lg:px-8">
             <h1 className="text-3xl font-bold tracking-tight text-gray-900 sm:text-4xl">Warenkorb</h1>
-
+            <hr />
+            <div>{JSON.stringify(cart)}</div>
+            <hr />
```
This debug information should be removed before merging...

no4any committed:
#48 fixed issues on update cart page

no4any committed:
#48 linter format

no4any committed:
Merge remote-tracking branch 'origin/main' into 48-order-page

khaoulaAmara committed:
#48 order page

eliasbnr reviewed (changes requested):
Please do not push any secrets; use the environment file instead. Instead of POST and GET, use NextActions. Also, do not push the .idea folder, as it contains project-specific settings for your IDE.

khaoulaAmara committed:
use NextAction and Environement variables

khaoulaAmara committed:
enhance comment

AWanly commented:
Merge conflict

khaoulaAmara committed:
add filter

khaoulaAmara committed:
feat: improve cart functionality, add toasts, translations, and full documentation

•⁠  ⁠added remove item functionality to shopping cart (with auto-removal when qty = 0)
•⁠  ⁠implemented toast popup to confirm add to cart
•⁠  ⁠added comprehensive JSDoc documentation across services and components
•⁠  ⁠improved UX for cart quantity handling and product feedback
